test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterHandle.java

```diff
@@ -0,0 +1,155 @@
+import org.elasticsearch.test.cluster.ClusterHandle;
+import org.elasticsearch.test.cluster.local.LocalClusterFactory.Node;
+import org.elasticsearch.test.cluster.local.model.User;
+import org.elasticsearch.test.cluster.util.ExceptionUtils;
+import org.elasticsearch.test.cluster.util.Retry;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinWorkerThread;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+public class LocalClusterHandle implements ClusterHandle {
+    private static final Logger LOGGER = LogManager.getLogger(LocalClusterHandle.class);
+    private static final Duration CLUSTER_UP_TIMEOUT = Duration.ofSeconds(30);
+
+    public final ForkJoinPool executor = new ForkJoinPool(
+        Math.max(Runtime.getRuntime().availableProcessors(), 4),
+        new ForkJoinPool.ForkJoinWorkerThreadFactory() {
+            private final AtomicLong counter = new AtomicLong(0);
+
+            @Override
+            public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+                ForkJoinWorkerThread thread = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+                thread.setName(name + "-node-executor-" + counter.getAndIncrement());
+                return thread;
+            }
+        },
+        null,
+        false
+    );
+    private final AtomicBoolean started = new AtomicBoolean(false);
+    private final String name;
+    private final List<Node> nodes;
+
+    public LocalClusterHandle(String name, List<Node> nodes) {
+        this.name = name;
+        this.nodes = nodes;
+    }
+
+    @Override
+    public void start() {
+        if (started.getAndSet(true) == false) {
+            LOGGER.info("Starting Elasticsearch test cluster '{}'", name);
+            execute(() -> nodes.parallelStream().forEach(Node::start));
+        }
+    }
+
+    @Override
+    public void stop(boolean forcibly) {
+        if (started.getAndSet(false)) {
+            LOGGER.info("Stopping Elasticsearch test cluster '{}', forcibly: {}", name, forcibly);
+            execute(() -> nodes.forEach(n -> n.stop(forcibly)));
+        } else {
+            // Make sure the process is stopped, otherwise wait
+            execute(() -> nodes.forEach(n -> n.waitForExit()));
+        }
+    }
+
+    @Override
+    public boolean isStarted() {
+        return started.get();
+    }
+
+    @Override
+    public void close() {
+        stop(false);
+
+        executor.shutdownNow();
+        try {
+            executor.awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
Is 5 seconds enough time? The node could be flushing state on shutdown right? Any reason to not wait the normal 30 seconds?

Reply:
We actually forcibly terminate now by default so this is actually _more_ lenient.